core/docker/default/etc/jvm.config

```diff
@@ -17,3 +17,5 @@
 -XX:+UseAESCTRIntrinsics
 # Disable Preventive GC for performance reasons (JDK-8293861)
 -XX:-G1UsePreventiveGC
+# Reduce starvation of threads by GClocker, recommend to set about the number of cpu cores (JDK-8192647)
+-XX:GCLockerRetryAllocationCount=32
```

Comment:
> update product test's jvm.config as well

I actually considered whether I should add in `testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/environment/multinode-all/jvm.config` as well at the beginning of our contribution~  

But I found that the heap space is 2G, which is shown below:  
![image](https://github.com/trinodb/trino/assets/26461591/33f818b0-8b40-4858-a8f4-68deed19e080)
So I thought that most Trino nodes have less CPU cores than the number of memory(mostly 1:4 or 1:6), so I thought that this config may be used to a node having 1 CPU core, so maybe it didn't need to add the parameter and keep the default value~

Reply:
i am not saying GCLockerRetryAllocationCount is needed for product tests
but, product tests jvm.config should be as close to the recommended settings as possible
so it will use lower Xmx by necessity, but GCLockerRetryAllocationCount should be same as recommended (unless there is a reason not to)